VideoLocker/src/main/java/org/edx/mobile/http/OauthRefreshTokenAuthenticator.java

```diff
@@ -53,14 +56,30 @@ public OauthRefreshTokenAuthenticator(Context context) {
     public Request authenticate(Route route, final Response response) throws IOException {
         logger.warn(response.toString());
 
-        if (!isTokenExpired(response.peekBody(HttpStatus.OK).string())) {
-            return null;
-        }
-
         final AuthResponse currentAuth = loginPrefs.getCurrentAuth();
         if (null == currentAuth || null == currentAuth.refresh_token) {
             return null;
         }
+        String response_body = response.peekBody(200).string();
+
+        switch (getErrorCode(response_body)) {
+            case TOKEN_EXPIRED_ERROR_MESSAGE:
+                break;
```

Comment:
We could do this if we didn't have the default case.

I added some doc strings.